src/btree/bt_page.c

```diff
@@ -426,12 +426,33 @@ __page_reconstruct_leaf_delta(WT_SESSION_IMPL *session, WT_REF *ref, WT_ITEM *de
                 tombstone->txnid = unpack.tw.stop_txn;
                 tombstone->start_ts = unpack.tw.stop_ts;
                 tombstone->durable_ts = unpack.tw.durable_stop_ts;
-                F_SET(tombstone, WT_UPDATE_DURABLE | WT_UPDATE_RESTORED_FROM_DELTA);
+                /*
+                 * Mark the update also as in-progress if the update and tombstone are from same
+                 * transaction by comparing both the transaction and timestamps as the transaction
+                 * information gets lost after restart.
+                 */
+                if ((unpack.tw.prepare && unpack.tw.start_ts == unpack.tw.stop_ts &&
+                      unpack.tw.durable_start_ts == unpack.tw.durable_stop_ts &&
```

Comment:
I think it is enough to just compare the txn id here. We can turn the comparison of timestamps to an assert.

Reply:
done
I have a question though, wouldn't it be just enough to check `unpack.tw.prepare` is true here (no need to compare transaction id) since we only enter this condition when `WT_TIME_WINDOW_HAS_STOP(&unpack.tw)` is true?

Reply:
You still need to check. We need to distinguish the case that only the stop is a prepare and the case that start and stop are both a prepare.

Reply:
so here we only handle the case that both start and stop are prepared, and ignore the case that only stop is prepared?